cmd/root.go

```diff
@@ -18,23 +19,23 @@ const exitError = 1
 // New adds all child commands to the root command and sets flags appropriately.
 func New() *cobra.Command {
 	cfg, err := config.Load("./meteor.yaml")
-	if err != nil {
+	if err != nil && !errors.As(err, &saltconfig.ConfigFileNotFoundError{}) {
 		fmt.Printf("ERROR: %s\n", err.Error())
-		os.Exit(1)
+		os.Exit(exitError)
 	}
 
-	lg := log.NewLogrus(log.LogrusWithLevel(cfg.LogLevel))
+	lgLvl := cfg.LogLevel
+	if lgLvl == "" {
+		lgLvl = "INFO"
+	}
+	lg := log.NewLogrus(log.LogrusWithLevel(lgLvl))
 	plugins.SetLog(lg)
 
 	// Setup statsd monitor to collect monitoring metrics
-	var mt *metrics.StatsdMonitor
-	if cfg.StatsdEnabled {
-		client, err := metrics.NewStatsdClient(cfg.StatsdHost)
-		if err != nil {
-			fmt.Printf("ERROR: %s\n", err.Error())
-			os.Exit(exitError)
-		}
-		mt = metrics.NewStatsdMonitor(client, cfg.StatsdPrefix)
+	mt, err := newStatsdMonitor(cfg)
+	if err != nil {
+		fmt.Printf("ERROR: %s\n", err.Error())
+		os.Exit(exitError)
```

Comment:
Rather than throwing `os.Exit(exitError)` here in `cmd/root`, I think it is better to handler `os.Exit` in `main.go` file. So maybe this New function could also return error and this line could just simply `return err`. wdyt? @sudo-suhas @StewartJingga

Reply:
Although this specific instance is no longer present, have done the suggested change for possible error during config load.